mobile_mhp/lib/services/notification_service.dart

```diff
@@ -0,0 +1,233 @@
+        FirebaseMessaging.onMessage.listen(_handleMessage);
+        FirebaseMessaging.onBackgroundMessage(
+            _firebaseMessagingBackgroundHandler);
+        FirebaseMessaging.onMessageOpenedApp.listen(_handleMessageOpenedApp);
+
+        // Fetch and print the FCM token
+        String? token = await _messaging.getToken();
+        _logger.i("FCM Token: $token");
+
+        // Fetch the APNS token for iOS
+        if (Platform.isIOS) {
+          String? apnsToken = await _messaging.getAPNSToken();
+          if (apnsToken != null) {
+            _logger.i("APNS Token: $apnsToken");
+          } else {
+            _logger.e("APNS token is null, notifications may not work on iOS.");
+          }
+        }
+      } else if (settings.authorizationStatus == AuthorizationStatus.denied) {
+        _logger.i('User denied notifications.');
+      }
+    } catch (e, stackTrace) {
+      _logger.e('Error initializing Firebase: $e');
+      await Sentry.captureException(e, stackTrace: stackTrace);
+    }
+  }
+
+  //* Local Notifications Initialization
+  Future<void> _initializeLocalNotifications() async {
+    const AndroidInitializationSettings initializationSettingsAndroid =
+        AndroidInitializationSettings('@mipmap/ic_launcher');
+
+    final DarwinInitializationSettings initializationSettingsIOS =
+        DarwinInitializationSettings();
+
+    final InitializationSettings initializationSettings =
+        InitializationSettings(
+      android: initializationSettingsAndroid,
+      iOS: initializationSettingsIOS,
+    );
+
+    await _flutterLocalNotificationsPlugin.initialize(
+      initializationSettings,
+      onDidReceiveNotificationResponse: (NotificationResponse response) {
+        selectNotificationSubject.add(response.payload);
+      },
+    );
+  }
+
+  //* Android Permissions Check
+  Future<void> _isAndroidPermissionGranted() async {
+    if (Platform.isAndroid) {
+      final bool granted = await _flutterLocalNotificationsPlugin
+              .resolvePlatformSpecificImplementation<
+                  AndroidFlutterLocalNotificationsPlugin>()
+              ?.areNotificationsEnabled() ??
+          false;
+
+      notificationsEnabled.add(granted);
+    }
+  }
+
+  //* Permissions
+  Future<void> requestPermissions() async {
+    if (Platform.isIOS || Platform.isMacOS) {
+      await _flutterLocalNotificationsPlugin
+          .resolvePlatformSpecificImplementation<
+              IOSFlutterLocalNotificationsPlugin>()
+          ?.requestPermissions(
+            alert: true,
+            badge: true,
+            sound: true,
+          );
+      await _flutterLocalNotificationsPlugin
+          .resolvePlatformSpecificImplementation<
+              MacOSFlutterLocalNotificationsPlugin>()
+          ?.requestPermissions(
+            alert: true,
+            badge: true,
+            sound: true,
+          );
+    } else if (Platform.isAndroid) {
```

Comment:
Since we are only targeting two platforms
```suggestion
    } else {
```